jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java

```diff
@@ -0,0 +1,184 @@
+        this.blacklistedPackagesList = blacklistedPackagesList;
+    }
+
+    /**
+     * Get All monitors and store metric metricClass to compositeData[].
+     *
+     * @throws OpenDataException creating compositetype fail
+     */
+    private void scan() {
+        final Set<Class<?>> monitors = new HashSet<>();
+        final ClassGraph classGraph = new ClassGraph();
+        classGraph.enableAnnotationInfo().ignoreClassVisibility().blacklistLibOrExtJars()
+                .removeTemporaryFilesAfterScan()
+                .overrideClassLoaders(this.getClass().getClassLoader()).ignoreParentClassLoaders();
+        if (whitelistedPackagesList.size() > 0) {
+            classGraph.whitelistPackages(whitelistedPackagesList.toString());
+        }
+        if (blacklistedPackagesList.size() > 0) {
+            classGraph.blacklistPackages(blacklistedPackagesList.toString());
+        }
+        final ScanResult scanResult = classGraph.scan();
+        final ClassInfoList classInfoList = scanResult.getClassesWithAnnotation(MetricClass.class.getName());
+        monitors.addAll(classInfoList.loadClasses(true));
+
+        final CompositeData[] monitorCompositeData = new CompositeData[monitors.size()];
+            int i = 0;
+        for (Class<?> monitor : monitors) {
+            try {
+                monitorCompositeData[i] = new MonitorCompositeData(monitor).getMetricAnnotation();
+                i++;
+            } catch (OpenDataException ox) {
+                logger.error("Getting monitor metricaAnnotation failed: " + monitor.getName());
+            }
+        }
+        final int monitorCount = monitorCompositeData.length;
+        final String[] monitorKey = new String[monitorCount];
+        final String[] monitorKeyDescription = new String[monitorCount];
+        @SuppressWarnings("rawtypes")
+        final OpenType[] monitorType = new OpenType[monitorCount];
+
+        for (int j = 0; j < monitorCompositeData.length; j++) {
+            final CompositeData monitor = monitorCompositeData[j];
+            monitorKey[j] = monitor.getCompositeType().getTypeName();
+            monitorKeyDescription[j] = monitor.getCompositeType().getDescription();
+            monitorType[j] = monitor.getCompositeType();
+        }
+
+        try {
+            final CompositeType allMonitorCompositeType = new CompositeType("Monitor Metric", "Monitor Metric Info", monitorKey,
+                    monitorKeyDescription, monitorType);
+            allMonitorCompositeData = new CompositeDataSupport(allMonitorCompositeType, monitorKey, monitorCompositeData);
+        } catch (final OpenDataException ox) {
+            logger.error("Creating CompositeData failed", ox);
+        }
+    }
+
+    /**
+     * Create CompositeData for existing monitor.
+     *
+     * @return monitorCompositeData all monitor CompositeData
+     */
+    @Override
+    public CompositeData getAllMonitorCompositeData() {
+        if (allMonitorCompositeData != null) {
```

Comment:
Can you make the if statement other way? "if (allMonitorCompositeData == null) { scan(); }